src/workspaces/features/ArchiveWorkspaceDialog/ArchiveWorkspaceDialog.tsx

```diff
@@ -4,49 +4,49 @@ import Spinner from "core/components/Spinner";
 import { useState } from "react";
 import { useTranslation } from "react-i18next";
 import { useRouter } from "next/router";
-import { useDeleteWorkspaceMutation } from "workspaces/graphql/mutations.generated";
+import { useArchiveWorkspaceMutation } from "workspaces/graphql/mutations.generated";
 import useCacheKey from "core/hooks/useCacheKey";
 import { gql } from "@apollo/client";
-import { DeleteWorkspace_WorkspaceFragment } from "./DeleteWorkspaceDialog.generated";
-import { DeleteWorkspaceError } from "graphql-types";
+import { ArchiveWorkspaceError } from "graphql-types";
+import { ArchiveWorkspace_WorkspaceFragment } from "./ArchiveWorkspaceDialog.generated";
 
-type DeleteWorkspaceDialogProps = {
+type ArchiveWorkspaceDialogProps = {
   onClose(): void;
   open: boolean;
-  workspace: DeleteWorkspace_WorkspaceFragment;
+  workspace: ArchiveWorkspace_WorkspaceFragment;
 };
 
-const DeleteWorkspaceDialog = (props: DeleteWorkspaceDialogProps) => {
+const ArchiveWorkspaceDialog = (props: ArchiveWorkspaceDialogProps) => {
   const router = useRouter();
   const { t } = useTranslation();
   const { open, onClose, workspace } = props;
   const [isSubmitting, setIsSubmitting] = useState(false);
 
-  const [deleteWorkspace] = useDeleteWorkspaceMutation();
+  const [archiveWorkspace] = useArchiveWorkspaceMutation();
   const clearCache = useCacheKey(["workspaces", workspace.slug]);
 
   const onSubmit = async () => {
     setIsSubmitting(true);
-    const { data } = await deleteWorkspace({
+    const { data } = await archiveWorkspace({
       variables: {
         input: {
           slug: workspace.slug,
         },
       },
     });
 
-    if (!data?.deleteWorkspace) {
+    if (!data?.archiveWorkspace) {
       throw new Error("Unknown error.");
     }
 
-    if (data.deleteWorkspace.success) {
+    if (data.archiveWorkspace.success) {
       clearCache();
       setIsSubmitting(false);
       router.push("/");
```

Comment:
It would be better to push the user to /workspaces (since we are going to add a feature flag to prevent the users to leave the workspace env)